reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -144,23 +143,39 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
     }
   }
 
-  private def storeSignedTarget[T](repoId: RepoId, signed: SignedPayload[TargetsRole]): Route = {
-    val f: Future[ToResponseMarshallable] = offlineSignedRoleStorage.store(repoId, signed).map {
-      case Valid(_) =>
-        StatusCodes.NoContent
+  private def storeSignedTarget[T](repoId: RepoId, namespace: Namespace, signed: SignedPayload[TargetsRole]): Route = {
+    onSuccess(offlineSignedRoleStorage.store(repoId, signed)) {
+      case Valid(newSignedRole) =>
+        newSignedRole.content.signed.as[TargetsRole].right.foreach { targetRole =>
```

Comment:
This could be written as `newSignedRole.content.signed.as[TargetsRole].foreach` in scala 2.12, or importing cats.syntax.either._ (we are moving to 2.12 anyway since rvi server is dead.

But I think this should probably be simplified to: 

```publishNewTargets(namespace, signed.signed.targets)``` 

because the `newSignedRole` version will fail silently if parsing fails. Using an already parsed object should be more robust.

Reply:
ah yeah that's better.